api/agent/protocol/json_test.go

```diff
@@ -38,10 +40,17 @@ func TestJSONProtocolDumpJSONRequestWithData(t *testing.T) {
 		Host: "localhost:8080",
 	}
 	var buf bytes.Buffer
-	rDataBefore := RequestData{A: "a"}
-	json.NewEncoder(&buf).Encode(rDataBefore)
+
+	if data != nil {
+		_ = json.NewEncoder(&buf).Encode(data)
```

Comment:
What happens if this is not able to encode / errors? shouldnt you handle?

Reply:
Well, data being encoded here is test data, it should not fail.

Reply:
Right. Gotcha 👍